docs/v8-migration.md

```diff
@@ -0,0 +1,471 @@
+        include: /\.vue$/,
+        presets: [
+          ['@babel/preset-typescript', {
+    -       allExtensions: true, isTSX: true
+    +       ignoreExtensions: true
+          }]
+        ]
+      }]
+    }
+    ```
+
+    If you want to preserve the JSX format but transpile the TypeScript part, use `ignoreExtensions: true` and add [`@babel/plugin-syntax-jsx`](./plugin-syntax-jsx.md) to `plugins`.
+
+  - `isTSX: false` and `allExtensions: true`
+
+    Use `ignoreExtensions: true`, see the example above.
+
+  - `isTSX: false` and `allExtensions: false`
+
+    You can safely remove them.
+
+
+![medium](https://img.shields.io/badge/risk%20of%20breakage%3F-medium-yellow.svg)
+
+- Remove `allowDeclareFields` option ([#12461](https://github.com/babel/babel/pull/12461))
+
+  **Migration**: Remove the option from your config. You will probably be fine with the new behaviour. Previously `allowDeclareFields` enables transforming the `declare` syntax introduced in TypeScript 3.7, in Babel 8 we support the syntax without such a flag. See also the [compilation changes](#compilation-change-jsx) section.
+
+- Type check input options ([#12460](https://github.com/babel/babel/pull/12460))
+
+  **Migration**: The preset will also report invalid option names. Refer to the [docs](./preset-typescript.md#options) and ensure valid usage. For example, `runtime` is not a valid `preset-typescript` option and thus should be removed.
+
+### `@babel/plugin-transform-typescript` {#configuration-change-transform-ts}
+
+![medium](https://img.shields.io/badge/risk%20of%20breakage%3F-medium-yellow.svg)
+
+- Remove `allowDeclareFields` option ([#12461](https://github.com/babel/babel/pull/12461))
+
+  **Migration**: Remove the option from your config.
+
+### `@babel/plugin-syntax-typescript` {#configuration-change-syntax-ts}
+
+![high](https://img.shields.io/badge/risk%20of%20breakage%3F-high-red.svg)
+
+- Remove `isTSX` option ([#14955](https://github.com/babel/babel/pull/14955))
+
+  **Migration**: If you are using `isTSX: true`, remove this option and add [`@babel/plugin-syntax-jsx`](./plugin-syntax-jsx.md) to `plugins`:
+
+  ```diff title="babel.config.json
+  {
+    "plugins": [
+  -   ["@babel/plugin-syntax-typescript", { "isTSX": true }]
+  +   ["@babel/plugin-syntax-typescript"]
+  +   ["@babel/plugin-syntax-jsx"]
+    ]
+  }
+  ```
+
+  If you are using `isTSX: false`, you can safely remove them.
+
+### `@babel/preset-flow` {#configuration-change-preset-flow}
+
+![medium](https://img.shields.io/badge/risk%20of%20breakage%3F-medium-yellow.svg)
+
+- Remove `allowDeclareFields` option ([#12457](https://github.com/babel/babel/pull/12457))
+
+  **Migration**: Remove the option from your config. You will probably be fine with the new behaviour. Previously `allowDeclareFields` enables transforming the `declare` syntax introduced in Flow 0.120.0, in Babel 8 we support the syntax without such a flag. See also the [compilation changes](#compilation-change-flow) section.
```

Comment:
```suggestion
  **Migration**: Remove the option from your config, since it's now enabled by default. Previously `allowDeclareFields` enables transforming the `declare` syntax introduced in Flow 0.120.0, in Babel 8 we support the syntax without such a flag. See also the [compilation changes](#compilation-change-flow) section.
```